webapp/components/form/Dropdown/Dropdown.js

```diff
@@ -1,143 +1,96 @@
 import './dropdown.scss'
-import React, { useCallback, useEffect, useRef } from 'react'
+
+import React from 'react'
 import PropTypes from 'prop-types'
-import ReactDOM from 'react-dom'
+import ReactSelect, { components } from 'react-select'
 import classNames from 'classnames'
 
-import * as A from '@core/arena'
-
 import { TestId } from '@webapp/utils/testId'
-import { Input } from '@webapp/components/form/Input'
-import AutocompleteDialog from '@webapp/components/form/AutocompleteDialog'
+import ValidationTooltip from '@webapp/components/validationTooltip'
+import { useDropdown } from './useDropdown'
+
+const OptionComponent = (props) => (
+  <div data-testid={TestId.dropdown.dropDownItem(props.data?.value)}>
+    <components.Option {...props} />
+  </div>
+)
 
-import ItemDialog from './ItemDialog'
-import { useLocalState, State } from './store'
+const IndicatorsContainerComponent = (props) => (
+  <div data-testid={TestId.dropdown.toggleBtn(props.selectProps.inputId)}>
+    <components.IndicatorsContainer {...props} />
+  </div>
+)
 
 const Dropdown = (props) => {
   const {
-    autocompleteDialogClassName,
     autocompleteMinChars,
     className,
+    clearable,
     disabled,
-    idInput,
+    id,
+    idInput: idInputProp,
     itemLabel,
     itemKey,
-    items,
+    items: itemsProp,
     onBeforeChange,
-    onChange,
+    onChange: onChangeProp,
     placeholder,
     readOnly,
     readOnlyInput,
     selection,
-    sourceElement,
+    testId,
     title,
     validation,
   } = props
 
-  const dropdownRef = useRef(null)
-  const inputRef = useRef(null)
-
-  const itemLabelFunction = useCallback(
-    (item) => (itemLabel.constructor === String ? A.prop(itemLabel, item) : itemLabel(item)),
-    [itemLabel]
-  )
-
-  const { state, Actions } = useLocalState({
-    autocompleteMinChars,
-    disabled,
-    itemKey,
-    itemLabelFunction,
-    items,
-    onBeforeChange,
-    onChange,
-    readOnly,
-    selection,
-    title,
-  })
-
-  // update itemLabelFunction and input value on itemLabelFunction change
-  useEffect(() => {
-    Actions.updateItemLabelFunction({ itemLabelFunction, selection })
-  }, [itemLabelFunction])
-
-  const showDialog = State.getShowDialog(state)
-  const itemsDialog = State.getItemsDialog(state)
-  const inputValue = State.getInputValue(state) || ''
-
-  const searchMinCharsReached = autocompleteMinChars <= 0 || inputValue.trim().length >= autocompleteMinChars
+  const { inputId, loading, menuIsOpen, onChange, onInputChange, openMenuOnClick, options, selectRef, value } =
+    useDropdown({
+      autocompleteMinChars,
```

Comment:
MinCharacters, It took me 3 seconds to guess what was Chars
or minCharactersToAutocomplete which is more descriptive

Reply:
yes, but maybe it would be even better if all this logic will be moved to another component, Autocomplete for example, that wraps the dropdown adding this autocompletion logic

Reply:
yes that could be better, in this case just change this small name makes a big difference

Reply:
ok